core/adapters/http_test.go

```diff
@@ -163,3 +188,82 @@ func TestHttpPost_Perform(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryParameters(t *testing.T) {
+	t.Parallel()
+
+	baseUrl := "http://example.com"
+
+	cases := []struct {
+		name        string
+		queryParams string
+		startingUrl	string
+		wantErrored bool
+		expected    adapters.QueryParameters
+	}{
+		{"empty", `""`, baseUrl, false, adapters.QueryParameters{}},
+		{
+			"array of params",
+			`["firstKey","firstVal","secondKey","secondVal"]`,
+			baseUrl,
```

Comment:
The startingUrl always appears to be the same, I would expect some with query parameters to make it clear how this PR affects query parameters that already exist in the URL.